tarpc/src/serde_transport.rs

```diff
@@ -433,22 +433,27 @@ pub mod unix {
     pub struct TempSock(std::path::PathBuf);
 
     impl TempSock {
-        /// A named socket that results in `<name>.sock`
+        /// A named socket that results in `<tempdir>/<name>.sock`
         pub fn new<S: AsRef<str>>(name: S) -> Self {
+            Self::exact(format!("{}.sock", name.as_ref()))
+        }
+
+        /// A named socket that uses the name exactly as passed without allocating a new `String`
+        /// to add `.sock`, this results in `<tempdir>/<name>`
+        pub fn exact<S: AsRef<str>>(name: S) -> Self {
             let mut sock = std::env::temp_dir();
-            sock.push(format!("{}.sock", name.as_ref()));
+            sock.push(name.as_ref());
             Self(sock)
         }
```

Comment:
this function should be easy enough to test? `assert!(temp_sock.ends_with(name));` or something?

Reply:
Added those tests along with some others including ensuring regular files are removed, or missing file errors are properly swallowed (such as if the `TempPathBuf` was never actually used to make a file/socket, Drop shouldn't panic).


[7da27a4](https://github.com/google/tarpc/pull/380/commits/7da27a45633c3d53f835d2a1aa2ac867dd297f8a)